app/models/user.rb

```diff
@@ -0,0 +1,52 @@
+class User < ActiveRecord::Base
+  has_many :reviews , class_name: "Review", foreign_key: "author_id"
+  belongs_to :foot_profile
+
+  def all_comments
+    # active record scopes
+  end
+
+  def upvote_score
+    # Vote.where()
+  end
+  # users.password_hash in the database is a :string
+
+  def password
+    @password ||= BCrypt::Password.new(password_hash)
+  end
+
+  def password=(new_password)
+    @password = BCrypt::Password.create(new_password)
+    self.password_hash = @password
+  end
+
+  def authenticate(password)
+    self.password == password
+  end
+
+  def shoes_with_reviews(user_id)
+    user = User.find(user_id)
```

Comment:
you should probably add some `.includes` statements to this query so you can get all the data you need in the least amount of queries.